ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/models/Type.java

```diff
@@ -51,4 +49,26 @@ public String getUnqualifiedName() {
     public String getNameUnderscores() {
         return helper.camelToUnderscore(getUnqualifiedName());
     }
+
+    public String generateMatcher() {
+        final StringBuilder outputBuilder = new StringBuilder();
+        final int numConstants = this.enumConstants.size();
+        for (int currentIndex = 0; currentIndex < numConstants; currentIndex++) {
+            outputBuilder.append(CHelper.indent(1));
+
+            if (currentIndex > 0) {
+                outputBuilder.append("} else ");
+            }
```

Comment:
I was using currentIndex here, to prepend the else case after the first go through.  I didn't see a way to do that by a property of Ds3EnumConstant